optaplanner-test/src/main/java/org/optaplanner/test/impl/score/stream/DefaultConstraintVerifier.java

```diff
@@ -37,8 +39,8 @@
 
     private final ConstraintProvider_ constraintProvider;
     private final SolutionDescriptor<Solution_> solutionDescriptor;
-    private ConstraintStreamImplType constraintStreamImplType = ConstraintStreamImplType.DROOLS;
-    private boolean droolsAlphaNetworkCompilationEnabled = !CoreComponentsBuilder.isNativeImage();
+    private ConstraintStreamImplType constraintStreamImplType;
+    private Boolean droolsAlphaNetworkCompilationEnabled;
```

Comment:
Because ConstraintVerifier has with methods (withDroolsAlphaNetworkCompilationEnabled() etc), I strongly believe it must behave similar to SolverConfig etc: so by default all fields are null and we don't use any primitive types for fields.
The defaults values (DROOLS, !isNativeImage) only kick when it builds something.

This way, it's possible to do withConstraintStreamImplType(solverConfig.get...().getConstraintStreamImplType()) even if the solverConfig.get...().getConstraintStreamImplType() returns null.

That in turn, means the spring (and hopefully soon the quarkus) code doesn't have to hard code the defaults too.

Reply:
I disagree. And especially for boolean. Boolean with 3 values is not a boolean.
OK for configs, but no to proliferation outside of configs.